src/api/middleware/utils.js

```diff
@@ -47,9 +47,39 @@ const or = (
     return next(new APIError(status_code, error_code, msg, { or: errors }));
 };
 
+/**
+ * Util to allow running conditionally a middleware
+ *
+ *
+ * @param {Function} verify: Function that returns a boolean indicating if the validator should be ran
+ * @param {Function} middleware: Express middleware to be run
+ * @param {object} Options:
+ *  - error_code: error code in case of error (default: ErrorTypes.VALIDATION_ERROR)
+ *  - msg: the message in case of error (default: ValidationReasons.UNKNOWN)
+ *  - status_code: The status used in the HTTP Response in case of error (default: BAD_REQUEST (400))
+ */
+const when = (
+    verify,
+    middleware,
+    {
+        error_code = DEFAULT_ERROR_CODE,
+        msg = DEFAULT_ERROR_MSG,
+        status_code = DEFAULT_STATUS_CODE
+    } = {}
+) => async (req, res, next) => {
+    if (verify(req)) {
```

Comment:
Allow for non-function to be passed as conditions as well ( update the comment above to reflect this as well)
```suggestion
    if ((typeof verify === "function" && verify(req)) || verify) {
```